staging/src/k8s.io/sample-controller/artifacts/examples/crd.yaml

```diff
@@ -0,0 +1,11 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: foos.example.controller.code-generator.k8s.io
```

Comment:
how about foos.sample-controller.k8s.io ?

Reply:
This name is invalid, as it'd result in the name of the group being `sample-controller`, which is invalid in Go package names. When building with this set, I get:

```
Generating clientset for example:v1alpha1 at k8s.io/sample-controller/pkg/client/clientset
F0925 21:59:07.859468   10700 main.go:202] Error: Failed executing generator: some packages had errors:
errors in package "k8s.io/sample-controller/pkg/client/clientset/versioned":
unable to format file "../../k8s.io/sample-controller/pkg/client/clientset/versioned/clientset.go" (24:8: expected 'STRING', found '-' (and 10 more errors)).

errors in package "k8s.io/sample-controller/pkg/client/clientset/versioned/scheme":
unable to format file "../../k8s.io/sample-controller/pkg/client/clientset/versioned/scheme/register.go" (22:8: expected 'STRING', found '-' (and 4 more errors)).
```

(with a load more, similar errors)

Reply:
Okay I've managed to actually get it to work by setting the groupName to not include the hyphen (`// +groupName=samplecontroller.k8s.io`) - this then creates random inconsistencies between using the `-` and not using it. What do you think of `foos.samplecontroller.k8s.io`?

Reply:
without the `-` is fine